File: project/core/lib/db/application/src/main/java/vdi/core/db/app/model/InstallStatus.java
package vdi.core.db.app.model;

public enum InstallStatus {

  /**
   * Dataset installation is currently in progress.
   */
  RUNNING("running"),

  /**
   * Dataset installation is successfully completed.
   */
  COMPLETE("complete"),

  /**
   * Dataset installation could not be completed due to a validation failure.
   */
  FAILED_VALIDATION("failed-validation"),

  /**
   * Dataset installation failed due to an internal or unexpected error.
   */
  FAILED_INSTALLATION("failed-installation"),

  /**
   * Dataset installation could not be completed due to one or more missing
   * dependencies.
   */
  MISSING_DEPENDENCY("missing-dependency"),

  /**
   * Dataset has been marked as ready for reinstallation.
   */
  READY_FOR_REINSTALL("ready-for-reinstall"),
  ;


  public final String value;

  InstallStatus(String value) {
    this.value = value;
  }

  public static InstallStatus fromString(String value) {
    for (var e : values())
      if (e.value.equals(value))
        return e;

    throw new IllegalArgumentException("unrecognized InstallStatus value: " + value);
  }
}

File: lib/db/application/src/main/kotlin/vdi/lib/db/app/model/InstallStatus.kt
package vdi.lib.db.app.model

/**
 * Dataset Installation Status
 *
 * Status value for the various states that a dataset installation could be in.
 *
 * @since 1.0.0
 *
 * @author Elizabeth Paige Harper - https://github.com/foxcapades
 */
enum class InstallStatus(val value: String) {

  /**
   * Dataset installation is currently in progress.
   */
  Running("running"),

  /**
   * Dataset installation is successfully completed.
   */
  Complete("complete"),

  /**
   * Dataset installation could not be completed due to a validation failure.
   */
  FailedValidation("failed-validation"),

  /**
   * Dataset installation failed due to an internal or unexpected error.
   */
  FailedInstallation("failed-installation"),

  /**
   * Dataset installation could not be completed due to one or more missing
   * dependencies.
   */
  MissingDependency("missing-dependency"),

  /**
   * Dataset has been marked as ready for reinstallation.
   */
  ReadyForReinstall("ready-for-reinstall"),
  ;

  companion object {
    @JvmStatic
    fun fromString(value: String): InstallStatus {
      for (enum in entries)
        if (enum.value == value)
          return enum

      throw IllegalArgumentException("unrecognized InstallStatus value: $value")
    }
  }
}

